zarrs/src/group.rs

```diff
@@ -245,8 +245,14 @@ impl<TStorage: ?Sized + ReadableStorageTraits + ListableStorageTraits> Group<TSt
     pub fn child_groups(&self) -> Result<Vec<Self>, GroupCreateError> {
         self.children()?
             .into_iter()
-            .filter(|node| matches!(node.metadata(), NodeMetadata::Group(_)))
-            .map(|node| Group::open(self.storage.clone(), node.name().as_str()))
+            .filter_map(|node| match node.metadata() {
+                NodeMetadata::Group(metadata) => Some(Group::new_with_metadata(
+                    self.storage.clone(),
+                    node.name().as_str(),
+                    metadata.clone(),
```

Comment:
Currently the metadata needs to be cloned, because the field is private so that destructuring of the `Node` is not possible.